src/realm/object-store/sync/sync_session.cpp

```diff
@@ -262,7 +262,10 @@ SyncSession::SyncSession(SyncClient& client, std::shared_ptr<DB> db, SyncConfig
         history.set_write_validator_factory(
             [weak_sub_mgr](Transaction& tr) -> util::UniqueFunction<sync::SyncReplication::WriteValidator> {
                 auto sub_mgr = weak_sub_mgr.lock();
-                REALM_ASSERT_RELEASE(sub_mgr);
+                if (!sub_mgr) {
+                    // A DB can outlive its sync session during client reset.
+                    return nullptr;
+                }
```

Comment:
@jbreams I ran into this assertion failing when pulling in the latest changes, can you verify that this is fine? The scenario is that the fresh Realm has its sync session torn down but the DBRef is passed along to the client reset mechanics. In recovery mode, client reset makes writes to the fresh Realm and doesn't care if there is a sync session attached because it doesn't commit anything. This is done to generate a sync changeset of the local writes  that have been recovered.

Reply:
Good idea. Updated to turn off the write validator when downloading the fresh Realm for client reset.